tests/scalers/couchdb/couchdb_test.go

```diff
@@ -0,0 +1,394 @@
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	// test scaling
+	testActivation(t, kc)
+	testScaleUp(t, kc)
+	testScaleDown(t, kc)
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func generateUUID() string {
+	id := strings.ReplaceAll(uuid.New().String(), "-", "")
+	return id
+}
+
+func installCouchDB(t *testing.T) {
+	_, err := ExecuteCommand(fmt.Sprintf("helm repo add couchdb %s", couchdbHelmRepo))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand("sudo helm repo update")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	uuid := generateUUID()
+	_, err = ExecuteCommand(fmt.Sprintf("sudo helm install test-release  --set couchdbConfig.couchdb.uuid=%s --namespace %s couchdb/couchdb", uuid, testNamespace))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+}
+
+func getPassword(kc *kubernetes.Clientset, namespace string) string {
+	secret, _ := kc.CoreV1().Secrets(namespace).Get(context.Background(), "test-release-couchdb", metav1.GetOptions{})
+	encodedPassword := secret.Data["adminPassword"]
+	password := string(encodedPassword)
+	fmt.Println("Secret Data: ", password)
+	return password
+}
+
+func WaitForDeploymentReplicaCountChangeCouchDB(t *testing.T, kc *kubernetes.Clientset, name, namespace string, iterations, intervalSeconds int) bool {
+	t.Log("Waiting for some time to see if deployment replica count changes")
+	var replicas, prevReplicas int32
+	prevReplicas = -1
+
+	for i := 0; i < iterations; i++ {
+		deployment, _ := kc.AppsV1().Deployments(namespace).Get(context.Background(), name, metav1.GetOptions{})
+		replicas = deployment.Status.Replicas
+
+		t.Logf("Deployment - %s, Current  - %d", name, replicas)
+
+		if replicas != prevReplicas && prevReplicas != -1 {
+			break
+		}
+
+		prevReplicas = replicas
+		time.Sleep(time.Duration(intervalSeconds) * time.Second)
+	}
+
+	return int(replicas) > 1
+}
+
+func deployPod(kc *kubernetes.Clientset, podName, args string) {
+	podSpec := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-couchdb-" + podName,
+			Namespace: testNamespace,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: "OnFailure",
+			Containers: []corev1.Container{
+				{
+					Name:    "test-couchdb-container",
+					Image:   "nginx:alpine",
+ 					Command: []string{"curl"},
+					Args:    strings.Split(args, " "),
+				},
+			},
+		},
+	}
+
+	_, err := kc.CoreV1().Pods(testNamespace).Create(context.Background(), podSpec, metav1.CreateOptions{})
+	if err != nil {
+			fmt.Println("Failed to create test pod: ", err)
+	}
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing activation ---")
+	password := getPassword(kc, testNamespace)
+	if WaitForStatefulsetReplicaReadyCount(t, kc, "test-release-couchdb", testNamespace, 3, 15, 1) {
```

Comment:
Can't we assume that CouchDB is ready? I mean, we should check if it's ready and wait for in during the installation process, not here

Reply:
No, it takes time to get ready, I have tested it